### PR TITLE
Add Notifications

### LINKED ISSUE #1505: Add Notifications
This is an issue meant to track the progress of the notifications feature.

Features I identified:
* [x] can be displayed by the app
* [x] can be closed by the user
* [x] can be closed programmatically
* [x] can be closed automatically after a set amount of time
* [x] can be updated
* [x] can be displayed by a plugin

Missing:
* [x] how they should look like (more like [this](https://ant.design/components/message/) or like [this](https://ant.design/components/notification/))

---

Related to https://github.com/camunda/camunda-modeler/issues/839

### PR DESCRIPTION
**Pull request by barmac**
Adds notifications appearing in the center below the toolbar.

__Which issue does this PR address?__

Closes #1505

### CODE REVIEW

File: client/src/app/App.js [hunk 1]
```diff
@@ -1092,6 +1096,10 @@ export class App extends PureComponent {
     });
   }
 
+  displayNotification(options) {
+    return this.notifications.current.displayNotification(options);
```
Comment (philippfromme) [inline client/src/app/App.js]: Shouldn't we pull out the notification logic out of the component that displays them so we could avoid using `current`.

Comment (barmac) [reply client/src/app/App.js]: Why should we?

**Comment by barmac** [comment]
I've added `cursor: pointer;` for the close button.

**Comment by barmac** [comment]
@philippfromme I fixed the overflow problem. It's ready for review.

**Comment by philippfromme** [comment]
@barmac Please stop pushing so I can finish the review. I fixed it already. 😅

**Comment by philippfromme** [comment]
> I've added `cursor: pointer;` for the close button.

I removed that. We don't want a desktop app to feel like a web page.

**Comment by philippfromme** [comment]
> @philippfromme I fixed the overflow problem. It's ready for review.

I had to remove the changes to the CSS, they made the canvas inaccessible.

**Pull request merged**